servers/shareable-lists-api/src/database/mutations/ShareableListItem.ts

```diff
@@ -104,18 +130,153 @@ export async function createShareableListItem(
   return listItem;
 }
 
+/**
+ * Mutation to bulk add items to an extant shareable list.
+ * Behaves like an 'append' action, ignoring any inputs that
+ * already exist in the list.
+ *
+ * @param context
+ * @param listId The "externalId" list identifier
+ * @param data
+ * @param userId
+ */
+export async function addToShareableList(
+  context: BaseContext,
+  { listExternalId, items }: { listExternalId: string; items: AddItemInput[] },
+  userId: number | bigint,
+): Promise<ListResponse> {
+  const { db, conn } = context;
+
+  // Retrieve the list this item should be added to.
+  // Note: no new items should be added to lists that have been taken down
+  // by the moderators.
+  // TODO (@kschelonka): revisit this visibility requirement with UX changes
+  // If the list doesn't exist it should fail from a FK constraint,
+  // but since the externalId isn't the FK we can't rely on that
+  const list = await db.list.findFirst({
+    where: {
+      externalId: listExternalId,
+      userId,
+      moderationStatus: ModerationStatus.VISIBLE,
+    },
+  });
+  if (!list) {
+    throw new NotFoundError(
+      `A list with the ID of "${listExternalId}" does not exist`,
+    );
+  }
+  // TODO: Kat - fix this input type, it's not the same as graphql schema
```

Comment:
```suggestion
  // TODO: (@kschelonka) - fix this input type, it's not the same as graphql schema
```

Reply:
Thank you, this I should have fixed/removed